gradle-plugins/src/main/java/io/opentelemetry/javaagent/muzzle/generation/MuzzleCodeGenerationPlugin.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.muzzle.generation;
+
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import java.net.URLClassLoader;
+import net.bytebuddy.build.Plugin;
+import net.bytebuddy.description.type.TypeDefinition;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.ClassFileLocator;
+import net.bytebuddy.dynamic.DynamicType;
+
+/**
+ * This class is a ByteBuddy build plugin that is responsible for generating actual implementation
+ * of some {@link InstrumentationModule} methods. Auto-generated methods have the word "muzzle" in
+ * their names.
+ *
+ * <p>This class is used in the gradle build scripts, referenced by each instrumentation module.
+ */
+public class MuzzleCodeGenerationPlugin implements Plugin {
```

Comment:
```suggestion
public final class MuzzleCodeGenerationPlugin implements Plugin {
```

Unless you know a use case for extending it